src/openforms/submissions/utils.py

```diff
@@ -90,7 +91,11 @@ def remove_submission_uploads_from_session(
 
 def send_confirmation_email(submission: Submission):
     logevent.confirmation_email_start(submission)
+    with translation.override(submission.language_code):
+        _send_confirmation_email(submission)
```

Comment:
imo the `translation.override` should only happen inside the `_send_confirmation_email` function body, before the `render_email_template` call(s).

Now you're also affecting the resulting language of the `logevent` calls inside the `_send_confirmation_email`, which should use the default language (ideally), otherwise admin users looking at submission logs will get mixed language content.

Probably good to make some assertions about these log events in the tests too.

Reply:
I found no clean way to test this in this test.
Looking at a `TimeLineProxy` instance it will always render in the active language at looking time, not at logging time.

Maybe it's an idea to have the logs ensure their output is in settings.LANGUAGE_CODE in a different ticket? I'm guessing production doesn't log to the database and testing on this TimeLineProxy is mute anyway.

Reply:
yeah I was confused - only the structure data that is used as context data is recorded, not the resulting message, that's rendered dynamically according to the active language of the user viewing the logs. Discard my comment!